libs/blocks/merch/merch.js

```diff
@@ -40,70 +27,88 @@ function omitNullValues(target) {
   return target;
 }
 
-const getTacocatEnv = (envName, locale) => {
-  const wcsLocale = GEO_MAPPINGS[locale.prefix] ?? locale.ietf;
-  // eslint-disable-next-line prefer-const
-  let [language, country = 'US'] = wcsLocale.split('-', 2);
+const imsCountryPromise = new Promise((resolve) => {
+  let count = 0;
+  const check = setInterval(() => {
+    count += 1;
+    if (window.adobeIMS) {
+      clearInterval(check);
+      if (window.adobeIMS.isSignedInUser()) {
+        window.adobeIMS
+          .getProfile()
+          .then(({ countryCode }) => resolve(countryCode))
+          .catch(() => resolve());
+      } else {
+        resolve();
+      }
+    } else if (count > 25) {
```

Comment:
Just curious how you came up with this number of retries? If I read this code correctly, you're trying get hold of `window.adobeIMS` with a 200ms wait interval in between for a max number of 25 retries = 5000ms. This seems a bit long imo.

Reply:
Until the Milo IMS supports is ready, I think this is a safe approach to wait for 5s.  It is not blocking so the UI thread.
Ideally, we will have a promise from milo core that will resolve as soon as imslibjs is loaded.